scripts/build_queryables.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+
+import json
+import os
+import requests
+
+STAC_API_ROOT = "https://planetarycomputer.microsoft.com/api/stac/v1"
+
+QUERYABLE_TEMPLATE = {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://example.org/queryables",
+    "type": "object",
+    "title": "",
+    "properties": {
+        "id": {
+            "title": "Item ID",
+            "description": "Item identifier",
+            "$ref": "https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/item.json#/definitions/core/allOf/2/properties/id"
+        },
+        "collection": {
+            "title": "Collection ID",
+            "description": "The ID of the STAC Collection this Item references to.",
+            "$ref": "https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/item.json#/definitions/core/allOf/2/properties/collection"
+        },
+        "geometry": {
+            "title": "Item geometry",
+            "description": "The footprint geometry for this Item.",
+            "$ref": "https://geojson.org/schema/Geometry.json"
+        },
+        "datetime": {
+            "description": "Datetime",
+            "type": "string",
+            "title": "Acquired",
+            "format": "date-time",
+            "pattern": "(\\+00:00|Z)$"
+        }
+    }
+}
+
+def collection_list():
+    """Collect a list of collection IDs"""
+    resp = requests.get(STAC_API_ROOT + "/collections")
+    data = resp.json()
+    collection_ids = [d["id"] for d in data['collections']]
+    return collection_ids
+
+def example_properties(collection_id: str):
+    """Get an item to use as an example of collection properties"""
+    resp = requests.get(STAC_API_ROOT + f"/search?collections={collection_id}&limit=1")
+    data = resp.json()
+    try:
+        properties = data["features"][0]["properties"]
+    except IndexError:
+        properties = {}
+    return properties
+
+def construct_schema(example_properties: dict):
+    """Construct a schema for provided properties, using the queryable template"""
+    template = QUERYABLE_TEMPLATE.copy()
+    for key, value in example_properties.items():
+        # pass over fields that are already included or are known to be useless in querying
+        if key in ["datetime", "description", "title"]:
```

Comment:
Here's an example of an existing queryable with some "handcrafted" properties (references to existing extension schemas and some with known values encoded as enums). Would be good to do a union of an existing queryable and the base template here so we can provide some overrides.

https://planetarycomputer-staging.microsoft.com/api/stac/v1/collections/sentinel-2-l2a/queryables